packages/core/src/hooks/useToken.ts

```diff
@@ -8,6 +8,9 @@ import { Contract } from 'ethers'
  * @public
  */
 export function useToken(tokenAddress: string | Falsy): TokenInfo | undefined {
+  if (!tokenAddress) {
```

Comment:
I don't think you can do that. Every call to the hook must invoke the same set of hooks in the same order every time. This one misses the `useCalls` below.

https://reactjs.org/docs/hooks-rules.html

Reply:
I think the result might be undefined regardless of this check.
Didn't test it though.

Reply:
@dmaretskyi you are right, knew about it but forgot to change. Now I've pushed updated code. Please check :)